### PR TITLE
add warning for BackupStorageRedundancy

### PR DESCRIPTION
**Pull request by xaliciayang**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs [hunk 1]
```diff
@@ -198,14 +205,38 @@ public class NewAzureSqlDatabase : AzureSqlDatabaseCmdletBase<AzureSqlDatabaseCr
         /// </summary>
         [Parameter(Mandatory = false,
             HelpMessage = "The Backup storage redundancy used to store backups for the SQL Database. Options are: Local, Zone and Geo.")]
-        [ValidateSet("Local", "Zone", "Geo")]
+        [ValidateSet("Local", "Zone", "Geo", IgnoreCase = false)]
```
Comment (msJinLei) [inline src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:208]: Change validate set IgnoreCase from true to false is a breaking change. Is it a bug fix? Will it report error from server when case mismatched?

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs [hunk 2]
```diff
@@ -198,14 +205,38 @@ public class NewAzureSqlDatabase : AzureSqlDatabaseCmdletBase<AzureSqlDatabaseCr
...
+                if (this.BackupStorageRedundancy == null)
+                {
+                    if (!Force.IsPresent && !ShouldContinue(
+                        string.Format(CultureInfo.InvariantCulture, Properties.Resources.DoYouWantToProceed, this.DatabaseName),
+                        string.Format(CultureInfo.InvariantCulture, Properties.Resources.GeoBackupRedundancyNotChosenWarning, this.DatabaseName)))
+                    {
+                        return;
+                    }
+                }
+                else if (string.Equals(this.BackupStorageRedundancy, "Geo", System.StringComparison.OrdinalIgnoreCase))
```
Comment (msJinLei) [inline src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:230]: Case is ignored here

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs [hunk 3]
```diff
@@ -198,14 +205,38 @@ public class NewAzureSqlDatabase : AzureSqlDatabaseCmdletBase<AzureSqlDatabaseCr
...
         public string BackupStorageRedundancy { get; set; }
 
+        protected static readonly string[] ListOfRegionsToShowWarningMessageForGeoBackupStorage = { "eastasia", "southeastasia", "brazilsouth", "east asia", "southeast asia", "brazil south" };
+
         /// <summary>
         /// Overriding to add warning message
         /// </summary>
         public override void ExecuteCmdlet()
         {
+            string location = ModelAdapter.GetServerLocation(ResourceGroupName, ServerName);
```
Comment (msJinLei) [inline src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:218]: - ModuleAdapter needs to be initialized before use( InitModelAdapter() is called in base.ExecuteCmdlet())
- ModelAdapter.GetServerLocation(ResourceGroupName, ServerName) will generate an additional call which requires re-recording of the related test cases.

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs (hunk 1 above)
Comment (xaliciayang) [reply src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:208]: This is a newly introduced paramater (as part of [PR 12899](https://github.com/Azure/azure-powershell/pull/12899)); @dingmeng-xue and I discussed this.  We introduced this to match the MI user experience ([PR 12849](https://github.com/Azure/azure-powershell/pull/12849)).

**Review by msJinLei — APPROVED** [review]

**Comment by dingmeng-xue** [comment]
Check fail because time out. Force merge it

**Pull request merged**